Automattic/newspack-newsletters
# feat: change usage report signature

### All Submissions:






### Changes proposed in this Pull Request:



A small refactor to the mailchimp usage report to

* Add the total number of contacts
* Add a Report class that will be shared by all ESPs to wirte their reports
* Drop the feature to look for reports more than one day old

See pamTN9-6YO-p2#comment-9637

### How to test the changes in this Pull Request:

With Mailchimp set as the ESP, run:
* `wp shell`
* `$rep = Newspack_Newsletters::get_service_provider())->get_usage_report();`
* `$rep->to_array()` and confirm you see the same results you see before this PR is applied but only with the data from yesterday. (see instructions in https://github.com/Automattic/newspack-newsletters/pull/1331) with the addition of `total_contacts`. 

### Other information:
